tests/acceptance/page_objects/base.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import absolute_import
+
+
+class BasePage(object):
+    page_name = 'base'
+
+    def __init__(self, browser):
+        self.browser = browser
+        try:
+            self.assert_correct_page()
+        except AssertionError:
+            raise Exception(
+                'This is not the %s page. Current url is %s.' %
+                (self.page_name, self.driver.current_url))
+
+    @property
+    def driver(self):
+        return self.browser.driver
+
+    def assert_correct_page(self):
+        raise NotImplementedError
+
+
+class BaseElement(object):
+    def __init__(self, selector, element):
+        self.selector = selector
+        self.element = element
+
+    def assert_appearance(self, *args, **kwargs):
+        pass
+
+
+class ButtonElement(BaseElement):
+    label_attr = 'aria-label'
+    disabled_attr = 'aria-disabled'
+
+    def assert_diabled(self, is_disabled=False):
+        disabled = 'true' if is_disabled else 'false'
+        assert self.element.get_attribute(self.disabled_attr) == disabled
```

Comment:
You might want to look at the DOM property for disabledness as it can be changed at runtime whereas the attribute is more static.